src/App.tsx

```diff
@@ -1,61 +1,135 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+const getUser = (id: number): User | null => {
+  return usersFromServer.find(user => user.id === id) || null;
+};
+
+const todoList: Todo[] = todosFromServer.map(todo => (
+  {
+    ...todo,
+    user: getUser(todo.userId),
+  }
+));
+
+const getTodoId = (todos: Todo[]): number => {
+  return Math.max(...todos.map(todo => todo.id)) + 1;
+};
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(todoList);
+  const [userError, showUserError] = useState(false);
+  const [titleError, showTitleError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    showTitleError(false);
+  };
+
+  const handleUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    showUserError(false);
+  };
+
+  const clear = () => {
+    showUserError(false);
+    showTitleError(false);
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!userId) {
+      showUserError(true);
+    }
+
+    if (!title.trim()) {
+      showTitleError(true);
+    }
+
+    if (userId && title.trim()) {
+      const newTodo: Todo = {
+        id: getTodoId(todos),
+        title,
+        completed: false,
+        userId,
+        user: getUser(userId),
+      };
+
+      setTodos(state => [...state, newTodo]);
+      clear();
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label htmlFor="title">Title: </label>
```

Comment:
Consider adding a `htmlFor` attribute to the label element to improve accessibility. 🌐